plugins/chat/app/serializers/base_chat_channel_membership_serializer.rb

```diff
@@ -8,5 +8,11 @@ class BaseChatChannelMembershipSerializer < ApplicationSerializer
              :chat_channel_id,
              :last_read_message_id,
              :unread_count,
-             :unread_mentions
+             :unread_mentions,
+             :can_join_chat_channel
+
+  def can_join_chat_channel
```

Comment:
@jjaffeux do you think there's a better way to achieve this without adding it to the serializer? I'm not 100% with the current solution.

Reply:
I would rather have it on the channel serializer

Reply:
Multiple things:
- first channel should be already loaded through `object.chat_channel` ? If not we should be careful to not add an N+1
- then it raises another question, why do we keep around a membership for a channel you can't join?

---

If we were to have a field like this I would probably have it in a meta attribute

Reply:
- Yes, the chat channel it's indeed being loaded through `object.chat_channel`. Missed that one.
- That's a really good point. Creating memberships only when needed makes much more sense if there not being used elsewhere (I don't think they are, anyway). Then we can just check if the record exists to avoid all this.

Thanks!

Reply:
What we are trying to fix here, is separate see from join, as in you can see a channel, but tou cant join it. If you cant join, you probably shouldnt have a membership I think.

So mostly public channels before loading should check for a see permission, if you cant see, nothing should arrive on frontend.